### ISSUE TITLE
App crash after applying proguard

### ISSUE THREAD

**Issue by hirenpatel868** [flashphoner/wcs-android-sdk-samples#5]
`W/System.err: java.lang.NoSuchMethodError: no non-static method "Lorg/webrtc/voiceengine/WebRtcAudioManager;.<init>(Landroid/content/Context;J)V"
E/rtc: #
    # Fatal error in ../../webrtc/modules/utility/source/helpers_android.cc, line 51
    # last system error: 0
    # Check failed: !jni->ExceptionCheck()
    # Error during GetMethodID: <init>, (Landroid/content/Context;J)V
    #
A/libc: Fatal signal 6 (SIGABRT), code -6 in tid 5790 (Thread-20), pid 5258 
Application terminated.`

**Comment by ehsanbhatti** [comment]
@hirenpatel868  did you found any solution to this problem? I am stuck at this from last 4 days and now I am out of options.

**Comment by hirenpatel868** [comment]
Didnt get any solutions..and no support from flashphoner developer team...left flashphoner and choose another vendor..

**Comment by ehsanbhatti** [comment]
@hirenpatel868 can you please share the vendor your using currently? I am looking for better alternative

**Comment by hirenpatel868** [comment]
Currently we are using AWS..

**Comment by ehsanbhatti** [comment]
@hirenpatel868 Thankyou for your response. By the way we have figureout the solution. We have added following rules in proGuard file :

-keep class com.flashphoner.** { *; }
-keep class org.webrtc.voiceengine.** { *; }
-keep class org.java_websocket.** { *; }

**Comment by hirenpatel868** [comment]
So is it working ???

**Comment by ehsanbhatti** [comment]
yes it is

**Comment by hirenpatel868** [comment]
Great...let me try again..thank you..

**Comment by dhameliyadharmesh** [comment]
After Release apk with Proguard. App Crashes 

My Proguard file content
-------------------------------------------
-keep class com.flashphoner.** { *; }
-keep class org.webrtc.voiceengine.* { *; }
-keep class org.java_websocket.* { *; }

app gradle file
---------------------------------
    apply plugin: 'com.android.application'

    android {
    compileSdkVersion 29
    defaultConfig {
        applicationId "com.xxxxxx"
        minSdkVersion 21
        targetSdkVersion 29
        versionCode 1
        versionName "1"
        vectorDrawables.useSupportLibrary = true
    }

    buildTypes {
        debug {
            minifyEnabled false
            shrinkResources false
            crunchPngs false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }

        release {
            minifyEnabled true
            shrinkResources true
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }
    }

    dependencies {

    // Required for androidx,third party dependencies
    implementation fileTree(include: ['*.jar'], dir: 'libs')  
implementation project(':wcs-android-sdk-1.1.0.5')
    }

**Comment by flashphoner** [comment]
> After Release apk with Proguard. App Crashes
> 
> ## My Proguard file content
> -keep class com.flashphoner.** { _; }
> -keep class org.webrtc.voiceengine._ { _; }
> -keep class org.java_websocket._ { *; }
> 

Please set `proguard-rules.pro` file content to the recommended [above](https://github.com/flashphoner/wcs-android-sdk-samples/issues/5#issuecomment-458493843):
```
-keep class com.flashphoner.** { ; }
-keep class org.webrtc.voiceengine.* { ; }
-keep class org.java_websocket.* { *; }
```

**Comment by dhameliyadharmesh** [comment]
@flashphoner Doesn't work for me
One more thing, I am releasing apk with **minifyEnabled true**
Showing me error while releasing apk


Expected char '}' at E:\xxx\xxx\xxx\app\proguard-rules.pro:40:34
-keep class com.flashphoner.** {  ;  }

**Comment by flashphoner** [comment]
@dhameliyadharmesh 
With the following proguard rules, the examples are build successfully
```
-keep class com.flashphoner.** { *; }
-keep class org.webrtc.** { *; }
-keep class org.java_websocket.** { *; }
-keep class androidx.** { *; }
-dontwarn **
```

**Comment by dhameliyadharmesh** [comment]
@flashphoner  Thank you! 
it's works. Also works on < 26 . I tested on 24 API level

**Comment by venkypokala9009** [comment]
**i am getting crashed every time. i have implemented in background service**

**wcs-android-sdk-1.1.0.13-release -- android- phone fuctions**

System.err: java.lang.IllegalArgumentException: Receiver not registered: org.webrtc.NetworkMonitorAutoDetect@dee728b
2020-06-21 22:35:03.079 28129-29090/com.evolgence.on W/System.err:     at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:1429)
2020-06-21 22:35:03.079 28129-29090/com.evolgence.on W/System.err:     at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1577)
2020-06-21 22:35:03.079 28129-29090/com.evolgence.on W/System.err:     at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:664)
2020-06-21 22:35:03.079 28129-29090/com.evolgence.on W/System.err:     at org.webrtc.NetworkMonitorAutoDetect.unregisterReceiver(NetworkMonitorAutoDetect.java:751)
2020-06-21 22:35:03.079 28129-29090/com.evolgence.on W/System.err:     at org.webrtc.NetworkMonitorAutoDetect.destroy(NetworkMonitorAutoDetect.java:729)
2020-06-21 22:35:03.079 28129-29090/com.evolgence.on W/System.err:     at org.webrtc.NetworkMonitor.stopMonitoring(NetworkMonitor.java:129)
2020-06-21 22:35:03.079 28129-29090/com.evolgence.on W/System.err:     at org.webrtc.NetworkMonitor.stopMonitoring(NetworkMonitor.java:138)
2020-06-21 22:35:03.079 28129-29090/com.evolgence.on E/rtc: #
    # Fatal error in: gen/sdk/android/generated_base_jni/jni/../../../../../../../../../usr/local/google/home/sakal/code/webrtc-aar-release/src/sdk/android/src/jni/jni_generator_helper.h, line 85
    # last system error: 0
    # Check failed: !env->ExceptionCheck()

**Comment by flashphoner** [comment]
@venkypokala9009 please create a topic on Flashphoner Support forum https://forum.flashphoner.com/
